pkg/plugins/manager.go

```diff
@@ -0,0 +1,106 @@
+package plugins
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	ErrPluginOfWrongType  = errors.New("plugin of the wrong type")
+	ErrPluginNameNotFound = errors.New("unknown plugin name")
+)
+
+var allowedProtocols = []plugin.Protocol{
+	plugin.ProtocolGRPC,
+}
+
+// PluginIdentity identifies the plugin's version and executable location.
+type PluginIdentity struct {
+	ProtocolVersion uint
+	// Cmd is the command that is used to run the plugin executable on the local filesystem.
+	Cmd exec.Cmd
+}
+
+// PluginHandshake includes handshake properties for the plugin.
+type PluginHandshake struct {
+	Key   string
+	Value string
+}
+
+// Manager maps the available plugin names to the different kinds of plugin.Client plugin controllers.
+// T is the custom interface type that the returned GRPCClient implementation implements, e.g. "Differ" for `plugin.Client`s that
+// include a GRPCClient that implements the "Differ" interface:
+// grpcPluginClient, err := c.Client() // Returns a plugin.GRPCClient
+// rawGrpcClientStub, err := grpcPluginClient.Dispense(name) // Calls grpcPluginClient's GRPCClient method and returns the gRPC stub.
+// grpcClient, ok := rawGrpcClientStub.(Differ) // Asserts the expected type of stub client.
```

Comment:
Consider:

```suggestion
//
// 	include a GRPCClient that implements the "Differ" interface:
// 	grpcPluginClient, err := c.Client() // Returns a plugin.GRPCClient
// 	rawGrpcClientStub, err := grpcPluginClient.Dispense(name) // Calls grpcPluginClient's GRPCClient method and returns the gRPC stub.
// 	grpcClient, ok := rawGrpcClientStub.(Differ) // Asserts the expected type of stub client.
```

which may produce a neater Godoc.